api/envoy/extensions/filters/listener/proxy_protocol/v3/proxy_protocol.proto

```diff
@@ -41,6 +43,15 @@ message ProxyProtocol {
     KeyValuePair on_tlv_present = 2;
   }
 
+  message TlvsMetadata {
```

Comment:
Seems weird to nest this inside configuration - AFAIU this is only used at runtime to represent typed metadata, not to configure the. filter. Should probably be top-level in this file.